### ISSUE TITLE
Validation needed on id_code for notification

### ISSUE THREAD

**Issue by kgf** [ReadyResponder/ReadyResponder#343]
Both Events and Notifications have id_codes. The validation would need to check both for duplicates. 
Id_code is optional for a notification, but if present, must not be a duplicate.
Also, an id_code can be recycled once the event/notification is cancelled/complete/closed. Even then, to prevent confusion, they should probably wait six months or so before being valid again.

**Comment by rhemaboyo** [comment]
Hi, can I go ahead and work on this issue?

**Comment by kgf** [comment]
@rhemaboyo That would be great, thanks !
Post here if you have any questions.
We appreciate the help.

**Comment by rhemaboyo** [comment]
For the event model I was thinking about validating uniqueness unless the most recent event with that id_code ended over 6 months ago. 

```
validates_uniqueness_of :id_code, :unless => :expired?
...
def expired?
    # 1.577e+7 = ~ 6months
    Event.where(:id_code => id_code).maximum('end_time') - Time.now + 1.577e+7 < 0
end
```

I was wondering if you want the notification model to have its own id_code or does it just share the id_code of the event it belongs to? (In which case the above validation would apply to both.)

**Comment by kgf** [comment]
A notification either has its own unique id code, or it has none.
Rails includes time helpers, so you can say `6.months.ago`. That would read better, I think.

**Comment by kgf** [comment]
@rhemaboyo We are quite close on this, hoping you'll be able to resolve the failing test.

**Comment by kgf** [comment]
Closed by #593
